grants_tagger/create_inclusion_list.py

```diff
@@ -0,0 +1,66 @@
+"""
+Script to automatically create a list of MeSH terms to include
+(i.e. filtering out the ones that have an attribute which mentions "Do not use")
+given a link to the MeSH xml file
+"""
+import typer
+
+import os
+import httpx
+import uuid
+import xml.etree.ElementTree as ET  # this is a library that can parse xml easily
+import pandas as pd
+from pathlib import Path
+
+
+def inclusion(xml_path, out_path):
+
+    # parse the xml tree
+    mesh_tree = ET.parse(xml_path)
+
+    # get the annotations and descriptors
+    annotations = []
+    descriptors = []
+    for mesh in mesh_tree.iter("DescriptorRecord"):
+        descriptors.append(mesh.find("DescriptorName").find("String").text)
+        annotation = mesh.find("Annotation")
+        if annotation is None:
+            annotations.append("")
+        else:
+            annotations.append(mesh.find("Annotation").text)
+
+    descriptors_to_use = []
+    descriptors_set = zip(descriptors, annotations)
+
+    # read in the list of descriptors we would like to drop manually:
+    exclusion_df = pd.read_csv(exclusion_list_path)
+    exclusion_list = list(exclusion_df["DescriptorName"].values)
+
+    # make a list of the descriptors we want to use
+    for descriptor, annotation in descriptors_set:
+        if (
+            "Do not use" not in annotation and descriptor not in exclusion_list
```

Comment:
Does "Do not use" always start with a capital D? If you wanted to make sure and catch all cases, you could check "do not use" against annotation.lower()

Reply:
It does. In fact, I am going to make this a little stronger and make it ": Do not use" as the examples the format is usually `INDEXER: Do not use` or `CATALOGER: Do not use`